lab_1_classify_by_unigrams/main.py

```diff
@@ -80,6 +158,28 @@ def preprocess_profile(profile: dict) -> dict[str, str | dict] | None:
     :return: a dict with a lower-cased loaded profile
     with relative frequencies without unnecessary ngrams
     """
+    if (
+            not isinstance(profile, dict)
+            or 'freq' not in profile.keys()
+            or 'name' not in profile.keys()
+            or 'n_words' not in profile.keys()
+    ):
+        return None
+
+    preprocessed_profile = {
+        'name': profile['name'],
+        'freq': {}
+    }
+
+    for token in profile['freq']:
+        if token.lower() in preprocessed_profile['freq']:
+            preprocessed_profile['freq'][token.lower()] \
```

Comment:
you call `lower` thrice - save it to a variable, also it is more readable if you save to variables  `profile['freq'][token]` and  `profile['n_words'][0]`

Reply:
done